token/transfer-hook-interface/README.md

```diff
@@ -0,0 +1,106 @@
+## Transfer-Hook Interface
+
+### Example program
+
+Here is an example program that only implements the required "execute" instruction,
+assuming that the proper account data is already written to the appropriate 
+program-derived address defined by the interface.
```

Comment:
If this is just an example program, shouldn't this be in a separate repo under the `solana-developers/` org?

Reply:
It defines the interface as well, which is used directly in token-2022. The example program is really just to help with integration testing, but this could be the right time to define how interfaces are declared here. How do you imagine packaging an interface? Does it just provide the state + instructions? Or does it also come with an example program for testing?

Reply:
Yeah I see what you're saying. Maybe just move the entrypoint & processor to `tests/` folder so its more clear.

What you're saying about testing makes sense. It's confusing because the lib is intertwined with the test program.